### PR TITLE
OmDet Turbo processor standardization

### PR DESCRIPTION
**Pull request by qubvel**
# What does this PR do?

Standardize post-processing for OmDet Turbo model (see #34926 for more info)

1) Rename `score_threshold` to `threshold` argument in `post_process_grounded_object_detection`
2) Rename `classes` to `text_labels` argument in `post_process_grounded_object_detection` and make it **optional**
3) Rename `classes` key to `text_labels` in post-processed output dictionary
4) Add `labels` (class indexes) key to post-processed output dictionary

The signature is going to be:
```python
    def post_process_grounded_object_detection(
        self,
        outputs: "OmDetTurboObjectDetectionOutput",
        text_labels: Optional[Union[List[str], List[List[str]]]] = None,           # <--------- renamed: classes -> text_labels + Optional now
        threshold: float = 0.3,                                                    # <--------- renamed: score_threshold -> threshold
        nms_threshold: float = 0.5,
        target_sizes: Optional[Union[TensorType, List[Tuple]]] = None,
        max_num_det: Optional[int] = None,
    )
```

The output is going to be:
```python
[
  {
      "boxes": torch.tensor of shape (N, 4),
      "labels": torch.tensor of shape (N,),                                       # <-------- new key, int indices, consistent with other object detection outputs
      "scores": torch.tensor of shape (N,),
      "text_labels": list of str names of len N (previously classes) or `None`    # <-------- renamed: classes -> text_labels
  },
  ...
]
```
"classes" key is also available, it will return `text_labels` and issue a warning

## TODO:

 - [x] Add deprecation for dict key

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qubvel** [comment]
cc @yonigozlan for review if you have bandwidth

File: src/transformers/models/omdet_turbo/processing_omdet_turbo.py [hunk 1]
```diff
@@ -290,66 +335,74 @@ def post_process_grounded_object_detection(
         Args:
             outputs ([`OmDetTurboObjectDetectionOutput`]):
                 Raw outputs of the model.
-            classes (Union[List[str], List[List[str]]]): The input classes names.
-            score_threshold (float, defaults to 0.3): Only return detections with a confidence score exceeding this
-                threshold.
-            nms_threshold (float, defaults to 0.5):  The threshold to use for box non-maximum suppression. Value in [0, 1].
-            target_sizes (`torch.Tensor` or `List[Tuple[int, int]]`, *optional*, defaults to None):
+            text_labels (Union[List[str], List[List[str]]], *optional*):
+                The input classes names. If not provided, classes will be set to `None` in `outputs`.
```
Comment (yonigozlan) [inline src/transformers/models/omdet_turbo/processing_omdet_turbo.py:339]:
```suggestion
                The input classes names. If not provided, `text_labels` will be set to `None` in `outputs`.
```
Unless I misunderstood.

**Review by yonigozlan — COMMENTED** [review]
Looks much better thanks for refactoring! 🤗

File: src/transformers/models/omdet_turbo/processing_omdet_turbo.py (hunk 1 above)
Comment (qubvel) [reply src/transformers/models/omdet_turbo/processing_omdet_turbo.py:339]: You are right 🤗

File: src/transformers/models/omdet_turbo/processing_omdet_turbo.py [hunk 2]
```diff
@@ -55,11 +69,23 @@ class OmDetTurboProcessorKwargs(ProcessingKwargs, total=False):
     }
 
 
-if is_torch_available():
-    import torch
+class _dict_with_warning(dict):
```
Comment (ArthurZucker) [inline src/transformers/models/omdet_turbo/processing_omdet_turbo.py:72]: let's use camel casing please!

File: src/transformers/models/omdet_turbo/processing_omdet_turbo.py [hunk 3]
```diff
@@ -290,66 +335,74 @@ def post_process_grounded_object_detection(
...
+                boxes=boxes,
+                scores=scores,
+                labels=batch_labels,
+                image_num_classes=image_num_classes,
+                image_size=image_size,
+                threshold=threshold,
+                nms_threshold=nms_threshold,
+                max_num_det=max_num_det,
             )
+            result = _dict_with_warning({"boxes": boxes, "scores": scores, "labels": labels, "text_labels": None})
```
Comment (ArthurZucker) [inline src/transformers/models/omdet_turbo/processing_omdet_turbo.py:400]: not 100% we need this (the class is a bit over engineered) but alright

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗 sorry for the delay

File: src/transformers/models/omdet_turbo/processing_omdet_turbo.py (hunk 2 above)
Comment (qubvel) [reply src/transformers/models/omdet_turbo/processing_omdet_turbo.py]: Fixed in https://github.com/huggingface/transformers/pull/34937/commits/2d7a6b24528d14c11bd6a3cd0b7b0a377374ab24
